src/config/mod.rs

```diff
@@ -66,12 +66,20 @@ pub fn parse_configs(
                 }
                 let parsed_network_config = parse_config(cfg.path().as_path())?;
 
+                let mut internal = false;
+
                 let network_name: String = match cfg.path().file_name() {
                     // This isn't *completely* safe, but I do not foresee many
                     // cases where our network names include non-UTF8
                     // characters.
                     Some(s) => match s.to_str() {
-                        Some(st) => st.to_string(),
+                        Some(st) => {
+			    let name_full = st.to_string();
+			    if name_full.ends_with(constants::INTERNAL_SUFFIX) {
+				internal = true;
+			    }
+			    name_full.strip_suffix(constants::INTERNAL_SUFFIX).unwrap_or(&name_full).to_string()
```

Comment:
Shouldn't `strip_suffix` be called on above bracket ( with `ends_with` ) only if a suffix was found ? Please ignore if this doesn't make sense.

Reply:
Oh i only meant that it performs the search again, but yeah this is looks safe and I hope compiler will automatically remove redundant search. No no issues I guess.